### PR TITLE
Support none encapsulation in AAC Parser

### PR DESCRIPTION
**Pull request by daniel-jodlos**
This Pull Request allows you to pass AAC Stream with `encapsulation: :none` on the input and request it to be packaged into ADTS

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -139,10 +139,10 @@ tags
...
 ### VisualStudioCode ###
 .vscode/*
-!.vscode/settings.json
-!.vscode/tasks.json
-!.vscode/launch.json
-!.vscode/extensions.json
+.vscode/settings.json
+.vscode/tasks.json
+.vscode/launch.json
+.vscode/extensions.json
```
Comment (mickel8) [inline .gitignore]: You sure? Do we want to commit e.g. `extensions.json`?

File: lib/membrane/aac/parser.ex [hunk 2]
```diff
@@ -4,15 +4,16 @@ defmodule Membrane.AAC.Parser do
 
   Supports both plain and ADTS-encapsulated output (configured by `out_encapsulation`),
   but currently accepts only ADTS AAC input.
+  Input with encapsulation :none is supported, but correct AAC caps need to be supplied with the stream.
```
Comment (mickel8) [inline lib/membrane/aac/parser.ex]: Those sentences should be merged

File: lib/membrane/aac/parser.ex [hunk 3]
```diff
@@ -36,15 +41,27 @@ defmodule Membrane.AAC.Parser do
...
+    case caps do
+      %AAC{encapsulation: :none} ->
+        {{:ok, caps: {:output, %{caps | encapsulation: state.out_encapsulation}}}, state}
+
+      _other ->
+        raise(
+          "%AAC{encapsulation: :none} caps are required when declaring in_encapsulation as :none"
+        )
+    end
+  end
```
Comment (mickel8) [inline lib/membrane/aac/parser.ex]: Maybe let's split this into two function clauses

File: lib/membrane/aac/parser.ex [hunk 4]
```diff
@@ -55,8 +72,28 @@ defmodule Membrane.AAC.Parser do
...
 
+  def handle_process(:input, buffer, ctx, state) when state.in_encapsulation == :none do
+    # Since there is no ADTS header, there is nothing to parse on the input
+    # Therefore, we only really add the timestamp
+    timestamp = Helper.next_timestamp(state.timestamp, ctx.pads.input.caps)
+    metadata = Map.put(buffer.metadata, :timestamp, timestamp)
+
+    buffer =
+      %{buffer | metadata: metadata}
+      |> then(&ensure_out_encapsulation(&1, ctx.pads.input.caps, state))
```
Comment (mickel8) [inline lib/membrane/aac/parser.ex]: Maybe 
```elixir
buffer = %{buffer | metadata: metadata}

buffer =
if state.out_encapsulation == :ADTS do
  %Buffer{buffer | payload: Helper.to_adts(caps, buffer.payload)}
else 
 buffer
end
``` 

I think it is easier to read than moving to a separate function.
Also ensure_out_encapsulation doesn't need to check if `in_encapsulation` is set to `:none` as we already checked this in function definition

File: lib/membrane/aac/parser.ex [hunk 5]
```diff
@@ -55,8 +72,28 @@ defmodule Membrane.AAC.Parser do
     end
   end
 
+  def handle_process(:input, buffer, ctx, state) when state.in_encapsulation == :none do
```
Comment (mickel8) [inline lib/membrane/aac/parser.ex:77]: Add `@impl true`

File: lib/membrane/aac/parser/helper.ex [hunk 6]
```diff
@@ -89,15 +89,59 @@ defmodule Membrane.AAC.Parser.Helper do
...
+  @spec next_timestamp(any(), AAC.t()) :: AAC.Parser.timestamp_t()
+  def next_timestamp(timestamp, caps) do
     use Ratio
 
     timestamp +
       Ratio.new(caps.samples_per_frame * caps.frames_per_buffer * Time.second(), caps.sample_rate)
   end
+
+  @spec to_adts(AAC.t(), binary()) :: binary()
+  def to_adts(%AAC{} = caps, payload) do
```
Comment (mickel8) [inline lib/membrane/aac/parser/helper.ex]: Maybe `payload_to_adts` or `add_adts_header` or something else that would be a little more descriptive.

Also, I would swithc `caps`  and `payload` so we would have `to_adts(payload, caps)` to be consistent with `next_timestamp`

**Review by mickel8 — APPROVED** [review]

File: lib/membrane/aac/parser.ex [hunk 7]
```diff
@@ -72,15 +71,23 @@ defmodule Membrane.AAC.Parser do
...
     buffer =
-      %{buffer | metadata: metadata}
-      |> then(&ensure_out_encapsulation(&1, ctx.pads.input.caps, state))
+      case state.out_encapsulation do
+        :ADTS ->
+          %Buffer{buffer | payload: Helper.payload_to_adts(buffer.payload, ctx.pads.input.caps)}
+
+        _other ->
+          buffer
+      end
```
Comment (mickel8) [inline lib/membrane/aac/parser.ex:92]: This is more `if else` than `case` as we have only two options

Comment (daniel-jodlos) [reply lib/membrane/aac/parser.ex:92]: There are more options, like LATM and I actually plan to include it, just not right now as we need this plugin to be released ASAP

**Pull request merged**